### PR TITLE
fix: `ListUsers` return correct relations in usersets

### PR DESCRIPTION
**Pull request by willvedd**
## Description
This PR fixes an issue in `ListUsers` where usersets weren't returning the relation in the response. Ex: (`group:fga` instead of `group:fga#member`). 

This bug occurred because we weren't checking the relation in the conditional when evaluating a userset _and_ because we weren't returning the correct object type. 

Good news: This fixes lots of tests
Bad news: This uncovers some edge cases about usersets

## Review Checklist
- [ ] I have clicked on ["allow edits by maintainers"](https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork).
- [ ] I have added documentation for new/changed functionality in this PR or in a PR to [openfga.dev](https://github.com/openfga/openfga.dev) [Provide a link to any relevant PRs in the references section above]
- [ ] The correct base branch is being used, if not `main`
- [ ] I have added tests to validate that the change in functionality is working as expected

### CODE REVIEW

File: assets/tests/consolidated_1_1_tests.yaml [hunk 1]
```diff
@@ -6148,9 +6143,9 @@ tests:
                 - organization#member
               object: repo:openfga/openfga
               relation: reader
-            temporarilySkipReason: "because userset relations not being returned properly, `organization:openfga` returned"
+            temporarilySkipReason: "because userset relations not being returned properly" # fails check, review if assertion is correct
             expectation:
-              - organization:openfga#member
+              - organization:openfga#member # is this correct?
```
Comment (miparnisari) [inline assets/tests/consolidated_1_1_tests.yaml:6148]: It's not. I came to the same conclusion in https://github.com/openfga/openfga/pull/1491/files

:)

File: pkg/server/commands/listusers/list_users_rpc_test.go [hunk 2]
```diff
@@ -84,7 +83,7 @@ func TestListUsersDirectRelationship(t *testing.T) {
 				tuple.NewTupleKey("group:eng", "member", "group:fga#member"),
 				tuple.NewTupleKey("group:fga", "member", "group:fga-backend#member"),
 			},
-			expectedUsers: []string{"group:fga#member", "group:fga-backend#member"},
+			expectedUsers: []string{"group:fga#member", "group:fga-backend#member", "group:eng#member"}, //Should `group:eng#member` be included here?
```
Comment (miparnisari) [inline pkg/server/commands/listusers/list_users_rpc_test.go:86]: after we merge https://github.com/openfga/openfga/pull/1066, yes

File: pkg/server/commands/listusers/list_users_rpc_test.go [hunk 3]
```diff
@@ -514,6 +510,70 @@ func TestListUsersUsersets(t *testing.T) {
...
+
+		  type org
+		    relations
+			  define member: [user]
+			  define admin: [org#member]
+
+          type repo
+            relations
+              define owner: [org]
+              define reader: admin from owner`,
```
Comment (miparnisari) [inline pkg/server/commands/listusers/list_users_rpc_test.go:537]: nit: please use tabs
